plugins/processors/ec2tagger/constants.go

```diff
@@ -0,0 +1,93 @@
+  # refresh_interval_seconds = 60
+  ##
+  ## Add tags for EC2 Metadata fields.
+  ## Supported fields are: "InstanceId", "ImageId" (aka AMI), "InstanceType"
+  ## If the configuration is not provided or it has an empty list, no EC2 Metadata tags are applied.
+  # ec2_metadata_tags = ["InstanceId", "ImageId", "InstanceType"]
+  ##
+  ## Add tags retrieved from the EC2 Instance Tags associated with this instance.
+  ## If this configuration is not provided, or has an empty list, no EC2 Instance Tags are applied.
+  ## If this configuration contains one entry and its value is "*", then ALL EC2 Instance Tags for the instance are applied.
+  ## Note: This plugin renames the "aws:autoscaling:groupName" EC2 Instance Tag key to be spelled "AutoScalingGroupName".
+  ## This aligns it with the AutoScaling dimension-name seen in AWS CloudWatch.
+  # ec2_instance_tag_keys = ["aws:autoscaling:groupName", "Name"]
+  ##
+  ## Retrieve ebs_volume_id for the specified devices, add ebs_volume_id as tag. The specified devices are
+  ## the values corresponding to the tag key "disk_device_tag_key" in the input metric.
+  ## If this configuration is not provided, or has an empty list, no ebs volume is applied.
+  ## If this configuration contains one entry and its value is "*", then all ebs volume for the instance are applied.
+  # ebs_device_keys = ["/dev/xvda", "/dev/nvme0n1"]
+  ##
+  ## Specify which tag to use to get the specified disk device name from input Metric
+  # disk_device_tag_key = "device"
+  ##
+  ## Amazon Credentials
+  ## Credentials are loaded in the following order
+  ## 1) Assumed credentials via STS if role_arn is specified
+  ## 2) explicit credentials from 'access_key' and 'secret_key'
+  ## 3) shared profile from 'profile'
+  ## 4) environment variables
+  ## 5) shared credentials file
+  ## 6) EC2 Instance Profile
+  # access_key = ""
+  # secret_key = ""
+  # token = ""
+  # role_arn = ""
+  # profile = ""
+  # shared_credential_file = ""
+`
+
+const (
+	ec2InstanceTagKeyASG = "aws:autoscaling:groupName"
+	cwDimensionASG       = "AutoScalingGroupName"
+	mdKeyInstanceId      = "InstanceId"
+	mdKeyImageId         = "ImageId"
+	mdKeyInstaneType     = "InstanceType"
+	ebsVolumeId          = "EBSVolumeId"
+)
+
+const (
+	metadataCheckStrStartInitialization         = "ec2tagger: EC2 IMDS has started initialization."
+	metadataCheckStrTagNotSupported             = "ec2tagger: Unsupported EC2 Metadata key: %s."
+	metadataCheckStrInstanceDocumentFailure     = "ec2tagger: Unable to retrieve Instance Metadata Tags."
+	metadataCheckStrEC2InstanceTagger           = "This plugin must only be used on an EC2 instance."
+	metadataCheckStrIncreaseHopLimit            = "ec2tagger: Please increase hop limit to 3 by following this document https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-options.html#configuring-IMDS-existing-instances."
```

Comment:
2 hops are recommended by [AWS](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html#imds-considerations). However, there are certain cases such as EKS Kops as @sethAmazon points out; moreover, I saw one use case of using 3 regarding hop limits when search on Slack Channel. 

Consider the security reason, 3 would not pose any impact as it would allow instance requests and the ones coming from containers while still blocking requests that might come from further away.

Reply:
The recommendation is 2 hops. There are specific scenarios where 3 hops is necessary but those are not typical. Is there any reason we would want to go against ECS recommendations for 2 hops for our general customer base?

Reply:
Redraw my statements regarding setting hop limit to 3. After setting hop limit to 2 with KOPs, I have been able to scrape metadata from IMDSv2. Therefore, I have updated my PR for changing recommendation to 2. However, would need everyone to agree on this before final decision.